### PR TITLE
XDR IOCs comments field

### PR DESCRIPTION
**Pull request by dorschw**
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-4549

### CODE REVIEW

File: Packs/CortexXDR/ReleaseNotes/4_9_4.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Cortex XDR - IOC
+- Added `the xsoar_comments_field` parameter.
```
Comment (dansterenson) [inline Packs/CortexXDR/ReleaseNotes/4_9_4.md:4]:
```suggestion
- Added the *xsoar_comments_field* integration parameter.
```

It would be helpful for customers if you could provide more information about this parameter's purpose.

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.yml [hunk 2]
```diff
@@ -165,12 +171,6 @@ script:
       required: false
       secret: false
     deprecated: false
-    description: Push modified IOCs to Cortex XDR.
-    execution: false
-    name: xdr-iocs-push
-  - arguments:
-    - default: false
-      description: The time of the file creation (use UTC time zone).
```
Comment (dansterenson) [inline Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.yml:173]: Why was this removed?

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py [hunk 3]
```diff
@@ -220,6 +221,21 @@ def demisto_types_to_xdr(_type: str) -> str:
         return xdr_type
 
 
+def _parse_demisto_comments(custom_fields: dict, comment_field_name: str) -> str | None:
```
Comment (dansterenson) [inline Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py:224]:
```suggestion
def _parse_demisto_comments(custom_fields: dict, comment_field_name: str) -> Optional[str]:
```
Don't forget to add `from typing import Optional`

**Comment by dansterenson** [comment]
Looks good, see my notes and please add the new parameter to the README file.

**Review by dansterenson — CHANGES_REQUESTED** [review]

File: Packs/CortexXDR/ReleaseNotes/4_9_4.md (hunk 1 above)
Comment (dorschw) [reply Packs/CortexXDR/ReleaseNotes/4_9_4.md:4]: I started describing it, but it became a bit too long for a release note (it's a bit complex), so I decided to keep it in the `description` part of the yaml instead. wdyt?

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.yml (hunk 2 above)
Comment (dorschw) [reply Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.yml:173]: not purposely (and the yaml is broken), fixing

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py (hunk 3 above)
Comment (dorschw) [reply Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py:224]: isn't it implicitly imported on CommonServerPython? 🤔

Comment (dansterenson) [reply Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py:224]: You are right

File: Packs/CortexXDR/ReleaseNotes/4_9_4.md (hunk 1 above)
Comment (dansterenson) [reply Packs/CortexXDR/ReleaseNotes/4_9_4.md:4]: What about something like this?:
```suggestion
- Added `the xsoar_comments_field` parameter to add tags to XDR IOCs.
```

Comment (dorschw) [reply Packs/CortexXDR/ReleaseNotes/4_9_4.md:4]: perhaps 
`- Added `the xsoar_comments_field` parameter, to choose the field used when pushing comments to XDR. The default remains *comments*.`

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py [hunk 4]
```diff
@@ -554,7 +586,8 @@ def main():  # pragma: no cover
         Client.tag = tag
     if xsoar_severity_field := params.get('xsoar_severity_field'):
         Client.xsoar_severity_field = to_cli_name(xsoar_severity_field)
-
+    if xsoar_comment_field := params.get('xsoar_comments_field'):
+        Client.xsoar_comments_field = xsoar_comment_field
```
Comment (dorschw) [inline Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py:590]:
```suggestion
        Client.xsoar_comments_field = xsoar_comment_field

```

**Comment by ShirleyDenkberg** [comment]
@dansterenson Doc review completed.

**Comment by Ni-Knight** [comment]
Looks good to me :)
